src/output-json.c

```diff
@@ -707,6 +710,70 @@ void CreateJSONFlowId(json_t *js, const Flow *f)
     }
 }
 
+static inline int JSONFormatAndAdd(json_t *parent, const char *key,
+                                   uint8_t *val, bool is_array) {
+    char eth_addr[19];
+    (void) snprintf(eth_addr, 19, "%02x:%02x:%02x:%02x:%02x:%02x",
+                    val[0], val[1], val[2], val[3], val[4], val[5]);
+    if (is_array) {
+        json_array_append_new(parent, json_string(eth_addr));
+    } else {
+        json_object_set_new(parent, key, json_string(eth_addr));
+    }
+    return 0;
+}
+
+/* only required to traverse the MAC address set */
+typedef struct JSONMACAddrInfo {
+    json_t *src, *dst;
+} JSONMACAddrInfo;
+
+static int JSONFlowAppendMACAddrs(uint8_t *val, int direction, void *data) {
+    JSONMACAddrInfo *info = (JSONMACAddrInfo*) data;
+    if (direction == TOSERVER) {
+        JSONFormatAndAdd(info->dst, NULL, val, true);
+    } else {
+        JSONFormatAndAdd(info->src, NULL, val, true);
+    }
+    return 0;
+}
+
+int CreateJSONEther(json_t *parent, const Packet *p, const Flow *f) {
+    json_t *ejs = json_object();
+    if (ejs == NULL) return 0;
+    if (p == NULL) {
+        /* we are creating an ether object in a flow context, so we need to
+           append to arrays */
+        if (f != NULL && f->macset && MacSetSize(f->macset) > 0) {
+            JSONMACAddrInfo info;
+            int ret;
+            info.dst = json_array();
+            info.src = json_array();
+            ret = MacSetForEach(f->macset, JSONFlowAppendMACAddrs, &info);
+            /* should not happen, JSONFlowAppendMACAddrs is sane */
+            if (ret != 0)
+                return ret;
+            json_object_set_new(ejs, "dst_macs", info.dst);
```

Comment:
For IP addresses we use `dest` instead of `dst`.  I think we should be consistent between the 2 (even if some of us prefer dst).

Reply:
Okay, sure.